packages/compiler-core/src/parse.ts

```diff
@@ -802,7 +803,7 @@ function parseText(context: ParserContext, mode: TextModes): TextNode {
   }
 
   let endIndex = context.source.length
-  for (let i = 0; i < endTokens.length; i++) {
+  for (let i = 0, len = endTokens.length; i < len; i++) {
     const index = context.source.indexOf(endTokens[i], 1)
```

Comment:
though , browser has jit, but it has a warm up process, always slower than cache the length

Reply:
If `endTokens` is a proxied array object, this modification is absolute yes.
If `endTokens` is a raw array object, I don't think this is necessary. The performance is the same but the latter introduces a new unnecessary variable `len`.

![image](https://user-images.githubusercontent.com/6134068/123031067-4829cc80-d416-11eb-88c0-7c9d7f6b63d1.png)